CONTRIBUTING.md

```diff
@@ -1,5 +1,123 @@
-# How to Contribute to Github (unfinished)
+# How to Contribute to this repository (unfinished)
 
-1. Step one
-  * Make sure to follow step one.
-2. Repeat first step.
\ No newline at end of file
+## Guidelines
+
+This year, we are using **git flow**, a development model to make our repository all neat and pretty!
+
+The main idea is to use separate branches for development and for features. The full blog post can be found [here](blogpost).
+
+### So... how do you use git flow?
+
+#### Installation
+
+##### The easy way
+
+- Install the git flow extension.
+    - Follow [these](instructions) instructions.
+    
+That's it! You're done! Well... ALMOST done. Now you just need to know how to use git flow. A cheat sheet can be found [here](cheatsheet)
+
+##### The "do it yourself" way
+
+The blog gives a good explanation on how to manually use commands to use the git flow technique. Although this is only recommended for people experienced in git, you can try it if you want. Just don't delete all our code. 
+
+## How to add code
+
+So let's say you have a great idea for the robot, such as better code for the shooter or something. How would you put this code on GitHub?
+
+DISCLAIMER: Again, this guide assumes that you have prior git knowledge. PLEASE learn git before you attempt this. You WILL break things if you don't.
+
+### 1. Get the code from GitHub
```

Comment:
Tell them to configure their ide and give a link to the gradlerio for dummies wiki post